src/hardware/serialport/softmodem.cpp

```diff
@@ -391,11 +395,22 @@ void CSerialModem::EnterConnectedState() {
 	CSerial::setRI(false);
 }
 
-void CSerialModem::DoCommand() {
+template <size_t N>
+bool starts_with(const char (&a)[N], const char *b) noexcept
```

Comment:
Only a suggestion, does not block the review.

I suggest renaming this function to : `peek_token` or `peek_next_token` or `is_next_token` to communicate, that this function is used in context of parsing?

```
if (is_next_token("NET", scanbuf) {
```

It can stay as it - we just have `starts_with` function in support already, but it does not fulfil requirements for this use-case, I think.

Reply:
Thanks @dreamer; went with `is_next_token` because it more specifically describes what we're trying to do and that we're looking for token - as opposed to simply doing string compares.  Did one last test, and all is good.

![2020-09-05_08-27](https://user-images.githubusercontent.com/1557255/92308356-dac5b180-ef51-11ea-9f27-284df4101cb7.png)